### PR TITLE
Fixed issue with bootloader GUI when in server_and_nuc deployed scenario

### PR DESCRIPTION
**Pull request by maxzieba**
## Proposed changes

User Story: https://shadowrobot.atlassian.net/browse/SP-299

Bootloader plugin has no warning about being launched on server laptop
Fixing issue when bootloader GUI ran on the server when using NUC.

If GUI launched on server in 'server and nuc' scenario you'll get the info below. If launched via RQT NUC, everything as usual. 
This doesn't interfere with server only scenario.

server_bootloader

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [X] Manually tested on hardware (if hardware specific or related).

## Documentation

- [X] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py [hunk 1]
```diff
@@ -94,45 +107,62 @@ def __init__(self, context):
...
-
-        self.server_revision = 0
-        self.diag_sub = rospy.Subscriber("/diagnostics", DiagnosticArray, self.diagnostics_callback)
+        if self._is_plugin_launched_on_correct_machine():
+            ui_file = os.path.join(rospkg.RosPack().get_path(
+                'sr_gui_bootloader'), 'uis', 'SrBootloader.ui')
+            loadUi(ui_file, self._widget)
+            self._widget.setObjectName('SrMotorResetterUi')
+
+            if context is not None:
```
Comment (jeferal) [inline advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py]: if not context? Like line 125

File: advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py [hunk 2]
```diff
@@ -306,7 +336,6 @@ def get_widget(self):
 
 
 if __name__ == "__main__":
-    rospy.init_node("bootloader")
```
Comment (jeferal) [inline advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py]: Why is this removed?

File: advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py (hunk 1 above)
Comment (maxzieba) [reply advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py]: I copied this part from the previous version, but I think this would be anyways 'if context:" ?

File: advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py (hunk 2 above)
Comment (maxzieba) [reply advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py]: Reverted (removed this accidentally after some tests)

**Review by georgiablanco — APPROVED** [review]

**Review by jeferal — APPROVED** [review]

**Review by bdenoun — APPROVED** [review]
